doc/adrs/152_reactivate_tool_selection_dialog.adoc

```diff
@@ -0,0 +1,144 @@
+      onClose={() => {
+        dispatch({ type: 'CLOSE_SELECTION_DIALOG' } as CloseSelectionDialogEvent);
+      }}
+      onFinish={(selectedObjectId) => {
+        dispatch({
+          type: 'HANDLE_SELECTED_OBJECT_IN_SELECTION_DIALOG',
+          selectedObjectId,
+        } as HandleSelectedObjectInSelectionDialogEvent);
+      }}
+    />
+  );
+}
+```
+
+The tool was finaly invoked within this useEffect (depending on the selectedObjectId value):
+
+```ts
+  useEffect(() => {
+    if (selectedObjectId && activeTool && contextualPalette) {
+      invokeTool(activeTool, contextualPalette.element.id, contextualPalette.startingPosition);
+      diagramServer.actionDispatcher.dispatch({ kind: SOURCE_ELEMENT_ACTION });
+      const resetSelectedObjectInSelectionDialogEvent: ResetSelectedObjectInSelectionDialogEvent = {
+        type: 'RESET_SELECTED_OBJECT_IN_SELECTION_DIALOG',
+      };
+      dispatch(resetSelectedObjectInSelectionDialogEvent);
+    }
+  }, [activeTool, diagramServer, invokeTool, dispatch, selectedObjectId, contextualPalette]);
+```
+
+== Decision
+
+The new diagram architecture based on react-flow has changed significantly. The tools are now invoked in the _Palette_ component:
+
+```ts
+  const handleToolClick = (tool: GQLTool) => {
+    switch (tool.id) {
+      case 'edit':
+        onDirectEditClick();
+        break;
+      case 'semantic-delete':
+        showDeletionConfirmation(() => {
+          invokeDelete(diagramElementId, GQLDeletionPolicy.SEMANTIC);
+        });
+        break;
+      case 'graphical-delete':
+        invokeDelete(diagramElementId, GQLDeletionPolicy.GRAPHICAL);
+        break;
+      case 'expand':
+        collapseExpandElement(diagramElementId, GQLCollapsingState.EXPANDED);
+        break;
+      case 'collapse':
+        collapseExpandElement(diagramElementId, GQLCollapsingState.COLLAPSED);
+        break;
+      default:
+        invokeSingleClickTool(tool);
+        break;
+    }
+  };
+```
+
+In a similar way than the delete tool shows the deletion confirmation dialog, the palette should call a showSelectionDialog if the tool defines a selectionDescriptionID.
+
+Palette.tsx:
+```ts
+  const { showDeletionConfirmation } = useDeletionConfirmationDialog();
+  const { showSelectionDialog } = useSelectionDialog();
+```
+
+The Palette will not call the invokeSingleClickTool directly but it will first verify if we need to show the selection dialog. The callback when performing the finish action will be responsible for invoking the tool with the selectedElementId.
+
+```ts
+if (tool.selectionDescriptionId) {
+  showSelectionDialog(editingContextId, tool.selectionRepresentationId, targetObjectId, onFinish, onClose);
+} else {
+  invokeSingleClickTool(tool);
+}
+
+```
+
+We will create a new hook "_useSelectionDialog_" in a similar way than the confirmationDialog. This hook will use the _SelectionDialogContext_.
+The _SelectionDialogContextProvider_ will be added in the DiagramRepresentation:
```

Comment:
We will not use a context to register the selection dialog but instead we will rely on the extension registry in order to retrieve the proper dialog to be displayed. We will have to see what we want from the extension point to retrieve the dialog. It could be a useData with both a component and a canHandle to indicate if the given dialogDescriptionId is supported